hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/key/OMKeyCommitRequest.java

```diff
@@ -158,6 +159,23 @@ public OMClientResponse validateAndUpdateCache(OzoneManager ozoneManager,
 
       validateBucketAndVolume(omMetadataManager, volumeName, bucketName);
 
+      // Check by the time we commit this file, is there any directory
+      // created if then fail it. This is being done for correctness.
+
+      // For now we are taking this approach similar to local file creation.
+      // TODO: In future this can be revisited to come up with an advanced
+      // approach.
+
+
+      if (ozoneManager.getEnableFileSystemPaths()) {
+        if (checkDirectoryAlreadyExists(volumeName, bucketName, keyName,
```

Comment:
Does the directory name have a trailing `/` in RocksDB? If so, do we need to append it to keyname before calling `checkDirectoryAlreadyExists`?

Reply:
>Does the directory name have a trailing / in RocksDB? 
Yes.
That is taken care of by the implementation of checkDirectoryAlreadyExists.

Where it uses getOzoneDirKey which appends "/"